CMakeLists.txt

```diff
@@ -303,9 +339,9 @@ IF (USE_SHA1DC)
 ELSEIF (WIN32 AND NOT MINGW)
 	ADD_DEFINITIONS(-DGIT_SHA1_WIN32)
 	FILE(GLOB SRC_SHA1 src/hash/hash_win32.c)
-ELSEIF (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+ELSEIF (TLS_BACKEND MATCHES "SecureTransport")
```

Comment:
This looks like it should use `${TLS_BACKEND}`, shouldn't it? Shouldn't this use `STREQUAL`, as all previous code? Same below for OpenSSL.